lib/streamlit/util.py

```diff
@@ -212,3 +212,21 @@ def extract_key_query_params(
             for item in sublist
         ]
     )
+
+
+def extract_single_element_lists(
+    dict: Dict[str, List[Any]]
+) -> Dict[str, Union[Any, List[Any]]]:
+    """
+    Extracts the single element from lists in a dictionary if the list has a length of 1.
```

Comment:
nit: summary of the docstring should go on the same line as the opening triple quotes